### ISSUE TITLE
Objects `null` properties throw TypeError

### ISSUE THREAD

**Issue by runeh** [sdgluck/mewt#32]
Steps to reproduce:

```
> immutable = require('mewt')
[Function]
> a = immutable({test: null})
TypeError: Cannot create proxy with a non-object as target or handler
    at new Proxy (<anonymous>)
    at module.exports (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:66:10)
    at k.reduce (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:60:11)
    at Array.reduce (native)
    at module.exports (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:58:24)
    at repl:1:5
    at ContextifyScript.Script.runInThisContext (vm.js:23:33)
    at REPLServer.defaultEval (repl.js:339:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
>
```

Not sure it this is supposed to work or not, but if it's disallowed, it would be nice to put it in the docs, and maybe throw a mewt error not a TypeError.

**Comment by sdgluck** [comment]
Hey- thanks for reporting this.

Looks like it is caused by my forgetting that `typeof null === 'object'`. D'oh.

Also in a nested call to `mewt`, which is what the library does for all elements of an array or properties of an object, I should be allowing primitive values. At the moment it would throw an error (with the above null check fixed) that `mewt` only accepts an array or object.

I'll look at putting a fix in asap. 👍

**Comment by runeh** [comment]
Cool. Thanks!

**Comment by sdgluck** [comment]
@runeh Fixed in v2.0.1 (see #34). Thanks again!
